### ISSUE TITLE
sticky: extensions

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [arkenfox/user.js#655]
previous threads #492  #294 #211 #12 

---

Use this issue for extension announcements: new, gone-to-sh*t, recommendations for adding or dropping in the wiki list [4.1: Extensions](https://github.com/ghacksuserjs/ghacks-user.js/wiki/4.1-Extensions). Stick to privacy and security related items

### :small_orange_diamond: possible additions

- [Site Bleacher](https://addons.mozilla.org/en-US/firefox/addon/site-bleacher/) | [GitHub](https://github.com/wooque/site-bleacher)

**Comment by Just-me-ghacks** [comment]
Any thoughts on [Trace](https://addons.mozilla.org/en-US/firefox/addon/absolutedouble-trace/) by AbsoluteDouble?

**Comment by StanGets** [comment]
Trace Firefox extension, IMHO, offers several of the features found on different extensions but doesn't really handle any of them correctly or at least as best as possible, as well as some others. Not yest anyway.

My feeling is that the developer's work scheme was to install practically all features right from the start, more or less elaborated (rather less) and progressively bring each of these components to maturity. My preference is rather to add new features only once those in place have been optimized, not before.

**Comment by Thorin-Oakenpants** [comment]
^^ https://github.com/ghacksuserjs/ghacks-user.js/issues/492#issuecomment-449595752

Let's not recover ground. Use the github search feature. The other thing is the visual design absolutely smacks of amateurism = a massive red flag for me = e.g it's like a novice who has discovered he can gradient colors. Another one is poor layout/structure and/or wasted space - not saying that is the case in this extension. Sign like this prom developers show a lack of years in the business - it takes a short while to become proficient in something, but a decade to become a wizard. 

I mean I haven't touched JS or CSS since 1999 (true story), although I do a lot of background and decades in other languages. And look at the fucking mess I made in [TZP](https://github.com/ghacksuserjs/TorZillaPrint) - not the layout or design (that comes from knowing the content and 20+ years of sql etc and report design), but the JS: I'm sure experienced web devs could look at my lame ass code and giggle like children, and point out flaws with var vs let, etc, and know of ways to speed things up.

To repeat: I wouldn't touch it with a barge pole - the extension, that is! TZP is fucking cool, I'd touch that with a barge pole .. for realsies :) I'd even let you guys touch it with your barge poles. Bargies all round!!

**Comment by crssi** [comment]
`window.opener be gone` can be removed from the list, its redundand with pref `2429` and this pref gives better protection

Add suggestions:
[HTTPZ](https://addons.mozilla.org/en-US/firefox/addon/httpz/) | [GitHub](https://github.com/claustromaniac/httpz/) -> `HTTPS by default` does not work when using `Temporary Containers`
[Privacy-Oriented Origin Policy]( https://addons.mozilla.org/firefox/addon/privacy-oriented-origin-policy/) | [GitHub](https://github.com/claustromaniac/poop/)

A nice TC companion:
[Context Plus](https://addons.mozilla.org/firefox/addon/context-plus/) | [GitHub](https://github.com/totallymike/contextPlus/)

**Comment by Thorin-Oakenpants** [comment]
Just to quickly reply about a couple of them

`HTTPS by default` is not trying to do what `HTTPZ` or `HTTPS Everywhere` is doing. It's for schemeless urls only. I was thinking of removing it from the wiki anyway - I used the web ext version of it for a while, ages ago, and it caused problems (can't exactly remember what - sometimes pages wouldn't load, from links, not schemeless ones, or there were delays etc - like it was interfering somehow). If HTTPS by default does not work with TC (please show me the report to the dev), then yeah, I can ditch it.

Kimetrak: I can see this info in uBO's dropdown and logger (and in UM)

**Comment by Thorin-Oakenpants** [comment]
re: window opener be gone: read https://github.com/ghacksuserjs/ghacks-user.js/issues/629

**Comment by StanGets** [comment]
> Kimetrak: I can see this info in uBO's dropdown and logger (and in UM)

Indeed. I have in mind another interesting Firefox extension which will as well provide the list of all sites accessed once on a page but will moreover display the security status of these connections:

![208853](https://user-images.githubusercontent.com/42912760/54112396-5939a580-43e6-11e9-9b92-4d99e3317414.png)

This is interesting because not provided by uBO.

**SixIndicator** on [AMO](https://addons.mozilla.org/en-US/firefox/addon/sixindicator/) and [GitHub](https://github.com/HostedDinner/SixIndicator)

**Comment by crssi** [comment]
About Kimetrac... I know you can see all that crap in uBO and uM and logger... but it caught my eye because:
uBO and uM here is blocking a lot of crap, but if something go 3rd party, then I can see "filtered" in Kimetrac and it helps me see much faster and easier, if there is something new that uBO and uM didn't block and worth to investigate to tighten uBO personal list.

That is it, nothing further to discuss anyway, since I have put it in my post under the "section": **Here on are not a suggestion, bust just for your info**... so I don't care. :smile: 

Cheers and :heart: you all :smile_cat:

**Comment by StanGets** [comment]
I've discovered a Firefox (& Chrome) extension which seems to me so worthy that I'd appreciate your opinion about it: **API-Killer-IndexedDB** at its [GitHub repository](https://github.com/eladkarako/chrome_extensions/tree/master/API-Killer-IndexedDB), available at [Add-ons for Firefox](https://addons.mozilla.org/en-US/firefox/addon/api-killer-indexeddb/).

What has always bothered me are sites laying data in my Firefox's profile storage/default folder, so called indexedDB. With this API-Killer-IndexedDB extension I can now avoid blocking cookie permission for sites such as youtube.com without having my indexedDB folder filled with unnecessary data (of course if cookie permission is session-only this data is removed on FF exit, yet I dislike sites laying on my computer what is not at all necessary).

Works great here. The developer has other extensions of which API-Killer-WebSocket and API-Killer-WebAssembly, all three in the scope of ghackuserjs concerns.

Any cons to argument?

**Comment by crssi** [comment]
@StanGets 
First and the only commit was done 5 hrs ago.
That is a fast discovery... or you know the author?

**Comment by StanGets** [comment]
@crssi I don't know the developer, I discovered the extension while reviewing AOM's updated extensions and immediately spotted API-Killer-IndexedDB because of the word killer associated to IndexedDB....

One thing is sure: it works. But as the developer notes it on his[ GitHub repository](https://github.com/eladkarako/chrome_extensions/tree/master/API-Killer-IndexedDB), "Kills HTML5' IndexedDB API, might break websites, if they do not have a localStorage/cookie fallback.". This is what I remain aware of but up to now, with cookies blocked and therefor indexedDB as well, I've encountered no problematic site.

I'm really enthusiastic about this extension but there may be cons, I'm no professional.

**Comment by Thorin-Oakenpants** [comment]
> of course if cookie permission is session-only this data [IDB] is removed on FF exit

Are you sure about that? Because IDB cannot be cleared by host. Here is an example of appCache not cleared by host: [862465](https://bugzilla.mozilla.org/show_bug.cgi?id=862465) - which they plan to nothing with because it's going to become obsolete.

You will only clear IDB after a session if 1) PB mode or 2) you clear "offline website data" on close (or manually with time range everything) or 3) Temp Containers

PS: I don't quite follow why by host can't be cleared, because each folder is labeled by domain - but there must be a technical reason. I also don't get why PB mode can't clear IDB on close if containers OA can be. PB mode is just another OA. Anyway - not my problem - just block those cookies is my motto :)

PPS: I haven't looked, so feel free to inspect that these extensions don't use any CSP header injection

**Comment by Thorin-Oakenpants** [comment]
FYI: https://nullsweep.com/launching-the-mozilla-plugin-privacy-test-database/

Grrr .. why oh why did he call them `plugins`

**Comment by StanGets** [comment]
Thanks for correcting me, @Thorin-Oakenpants :

> You will only clear IDB after a session if 1) PB mode or 2) you clear "offline website data" on close (or manually with time range everything) or 3) Temp Containers

Indeed I have set Firefox to clear "offline website data" on close. Wow, I had it all wrong, thanks agaiin.

> PPS: I haven't looked, so feel free to inspect that these extensions don't use any CSP header injection

I'm afraid that's above my skills. I mentioned the extension because it solves my problems on websites where i'd like to have a cookie -- i.e. YouTube when a userscript aiming to block Autoplay does it by modifying the site's cookie -- but where allowing the cookie would have that site lay itself in my IDB ... but considering the best often includes drawbacks is why I ask here advice.

**Comment by Thorin-Oakenpants** [comment]
I kinda fail to see the point, esp if you use FPI. If you allow a cookie on a site so it works properly (or you need to log in), then the clear on close does the same job - no other 1st party can get the data, and even with no FPI, no other site can get the data if you have 3rd party cookies blocked, let alone 1st party cookies. Does that make sense

eg - you allow cookies on YT, so you get a YT-idb, YT-cookie etc
eg - you load your fav site with embedded YT vids, no 3rd party cookies (or 1st), so no data is sent anywhere

If you're allowing the cookie, then you're allowing repeat 1st party visits per session, so IDB hardly makes a diff (I guess some sites only use IDB?, in which case, why do you need to enable cookies but block IDB). I think you're just over-complicating it :)

**Comment by StanGets** [comment]
About the Mozilla Plugin Privacy Test Database,
> 
Grrr .. why oh why did he call them plugins

Indeed! I know many people who still mistake add-ons with extensions with plugins ...
Another interesting topic and dedicated article. More you dig more astonishing revelations appear. I'm bookmarking that page.

**Comment by StanGets** [comment]
> I kinda fail to see the point, esp if you use FPI.
Yeah, you're absolutely right @Thorin-Oakenpants , and I do use FPI!
I'll be frank, I'm overdoing it, not for sentimental reasons but basically for psychological ones, in other terms even if other settings do the job I insist on extras even if they appear to not at all be implied in enhanced privacy for the sole reason of a non-rational principle : I don' t like sites writing to my device unless I've authorized them to. But you are right, it is not necessary. Maybe am I getting obsessed? LOL

**Comment by Thorin-Oakenpants** [comment]
> Maybe am I getting obsessed? LOL

I hear ya, brother! Still, it better than being possessed :ghost:

**Comment by StanGets** [comment]
> I hear ya, brother! Still, it better than being possessed

Beware, @Thorin-Oakenpants , you're pushing me to my natural behavior of being talkative on an off-topic : being obsession versus being possessed. I can make 200 lines on that! Two should be enough? OK :

As always the right point is the balanced one. Here at ghacksuserjs this is obviously the approach and that's good. It's good and even required in scientific/technological areas but maybe always, even if feelings wouldn't always fit in a hyper-rational way of conducting our thoughts.

Being obsessed may be a form of being possessed, by ourselves, by our unconscious, by fears but also by domination : the point is not to dominate but to control. Self-control is known, self-domination would be an awkward advice!

Maybe not two lines, I needed a minimum of two paragraphs and still... I controlled myself, lol!

**Comment by Thorin-Oakenpants** [comment]
@StanGets your bug [1525917](https://bugzilla.mozilla.org/show_bug.cgi?id=1525917) just landed - don't forget to make sure it works on FF68

**Comment by StanGets** [comment]
I received simultaneously your comment and bugzilla's via email notification.
I'm happy the issue became a concern for Bugzilla to start with, but I had only submitted naively the remark of a non-techie concerning the cookie behavior not respected in some situations (as i felt it); from there on the developments which lead to what seems to be a true correction today (applicable in FF68) flies 40,000 feet above me.
I'm a final release user so I'll have to wait for FF68 release to appreciate the work! It'll be then my moment of fame, people will say "if it hadn't been for Stangetz where would be now with this issue?" and I"ll reply "Oh! you know, I just pointed out, lucky I was as players who play for the first time...".

There I go again. Tea was too strong :=)

**Comment by crssi** [comment]
@StanGets 

> I'm afraid that's above my skills.

Its actually simple to do. See the last line in the post https://github.com/ghacksuserjs/ghacks-user.js/issues/664#issue-420621480

1. Install extension [CRX](https://addons.mozilla.org/firefox/addon/crxviewer/)
2. Open https://addons.mozilla.org/firefox/addon/api-killer-indexeddb/
3. Click on yellow `CRX` icon on the right side of URL bar and then `View source`
4. Enter `!content-security-policy` into the search field (upper left corner). NOTE: `!` means search all files.
5, If you get a hit, then most probably the extension is modifying the CSP.

Cheers :)

**Comment by StanGets** [comment]
@crssi thanks! Done and imputing !content-security-policy led to 0 hits.

But what I don't understand is the CRX extension being a requirement for checking CSP. Can't I just download an extension's xpi file, unzip it and search from there on? Second point is, how is searching for `content-security-policy` performed? Does CRX search for a specific term or specific code? Because if the query is only `content-security-policy` then I could as well search for it from the unzipped xpi ... 

Anyway, thanks. This is not school, forget my wondering...

**Comment by crssi** [comment]
> But what I don't understand is the CRX extension being a requirement for checking CSP.

No. Its not, but makes the whole process much much simpler. For sure you can just download and unzip, which CRX essentially is doing already for you. :wink: 
API for CSP is called over `content-security-policy`, so if not found then CSP does not get modified. If found, then you need to review the code in those lines.

Cheers

**Comment by Kraxys** [comment]
> 
> 
> previous threads #492 #294 #211 #12
> 
> Use this issue for extension announcements: new, gone-to-sh*t, recommendations for adding or dropping in the wiki list [4.1: Extensions](https://github.com/ghacksuserjs/ghacks-user.js/wiki/4.1-Extensions). Stick to privacy and security related items
> ### small_orange_diamond possible additions
> 
>     * [Site Bleacher](https://addons.mozilla.org/en-US/firefox/addon/site-bleacher/) | [GitHub](https://github.com/wooque/site-bleacher)
> 
> 
> ### small_orange_diamond nah

I find Site Bleacher interesting because it seems to handle IndexedDB in a more clever way than other comparable addons. For what I have seen, the IndexedDB a site has put in my browser, while remaining after closing my tab, is cleared as soon as I'm visiting this site again. This seems to me to be the most efficient way to handle IDB, given the API limitation.

**Comment by Thorin-Oakenpants** [comment]
> while remaining after closing my tab

The user.js clears it by default. So I assume you override this pref, and prefer to keep some IDB data by using a whitelist or something in site bleacher?

**Comment by AMihail1** [comment]
Thoughts on https://github.com/claustromaniac/poop ?

Will this be another CSP nightmare type extension?

**Comment by crssi** [comment]
^^ This extension doesn't even touch CSP. Did you even check?

**Comment by atomGit** [comment]
@StanGets with regard to CRX asked...

> Second point is, how is searching for content-security-policy performed?

in the CRX search input use:
`!content-security-policy`
the exclamation char prefix tells CRX to look at the content of the source files (default is file name) - i also use this to search for 'http' ( `!http` ) to look at URLs

**Comment by atomGit** [comment]
re: Site  Bleacher - been using it for a while and, according to dev, it does not raise entropy (he's not injecting anything into IDB storage that website can read like i thought he may have been)

i just asked him if it handles Workers cache, but i'm pretty sure it don't

**Comment by Thorin-Oakenpants** [comment]
AFAICT all your api-killer stuff has been removed from AMO

**Comment by StanGets** [comment]
> AFAICT all your api-killer stuff has been removed from AMO

The developer has removed all his API-Killers and all his other extensions removed except one, or thse have been removed by Mozilla, no idea.

I had indeed mentioned the API-Killer-IndexedDB for the reasons evoked here above. The extension having been removed from AMO, and because I ignore for what reasons, I've removed it as well from my Firefox profile.

Because I continue to dislike sites pouring data in my IDB, I've found another way to block the IDB Web Api : [WebAPI Blocker](https://addons.mozilla.org/en-US/firefox/addon/webapi-blocker/)

I checked all occurrences of IDBxxx proveded by this WebAPI blocker and disabled all 14 of them, which are:

IDBCursor
IDBCursorWithValue
IDBDatabase
IDBFactory
IDBFileHandle
IDBFileRequest
IDBIndex
IDBKeyRange
IDBMutableFile
IDBObjectStore
IDBOpenDBRequest
IDBRequest
IDBTransaction
IDBVersionChangeEvent

Works like a charm. Certainly not all 14 need to be disabled but until I check the ones strictly required i disable all. No issues at this time.

**Comment by atomGit** [comment]
so... i asked the Site Bleacher dev if he would have a go at cleaning the 'service workers' stuff and he did :)

in addition to cookies, local storage and IndexedDB, the extension also addresses service workers, cache storages, filesystems and webSQLs - i don't know exactly what's covered by the latter 3, so i asked him [here](https://github.com/wooque/site-bleacher/issues/5) if anyone cares to follow that and his answer was "Don't really know"

**Comment by jingofett** [comment]
Is there a downside to using Clean Links over the other link cleaners listed on the wiki?

https://addons.mozilla.org/en-US/firefox/addon/clean-links-webext/

Personally, I find this extension catches and cleans a lot more links than the alternatives (ClearURLs, Neat URL, Skip Redirect), but I remember back before webextensions, people having an issue with it. 

I use it with the following settings:

![image](https://user-images.githubusercontent.com/8313249/58339274-bd420680-7e38-11e9-923b-34a57d9e7f9d.png)

**Comment by atomGit** [comment]
> Is there a downside to using Clean Links  ...

somebody more knowledgeable might chime in, but IMO CleanURLs is the best of the bunch because it covers more and breaks less (not sure i've ever ClearURLs break anything) - it's been an install & forget ext. for me - no need to fiddle with white/black lists (doen't even have one)

some may not like it because it uses an external file (hosted on gitlab) but that's actually a plus in one way in that the dev doesn't have to update the ext. every time they need to change something

**Comment by jingofett** [comment]
> 
> 
> > Is there a downside to using Clean Links  ...
> 
> somebody more knowledgeable might chime in, but IMO CleanURLs is the best of the bunch because it covers more and breaks less (not sure i've ever had ClearURLs break anything) - it's been an install & forget ext. for me - no need to fiddle with white/black lists (doen't even have one)
> 
> some may not like it because it uses an external file (hosted on gitlab) but that's actually a plus in one way in that the dev doesn't have to update the ext. every time they need to change something

When using the examples on this page to test:
https://github.com/tumpio/requestcontrol/wiki/Testing-links

Clean Links successfully cleans most of them, except for the "no redirection, only parameters" group (except for example #11) and #14 in misc. ClearURLs cleans: #2, #6, #7, #8, #11

Again, I'm not an expert on this but I'm just asking so I can get more information

**Comment by jingofett** [comment]
> 
> 
> > Is there a downside to using Clean Links  ...
> 
> somebody more knowledgeable might chime in, but IMO CleanURLs is the best of the bunch because it covers more and breaks less (not sure i've ever had ClearURLs break anything) - it's been an install & forget ext. for me - no need to fiddle with white/black lists (doen't even have one)
> 
> some may not like it because it uses an external file (hosted on gitlab) but that's actually a plus in one way in that the dev doesn't have to update the ext. every time they need to change something

When using the examples on this page to test:
https://github.com/tumpio/requestcontrol/wiki/Testing-links

Clean Links successfully cleans most of them, except for the "no redirection, only parameters" group (except for example no.11) and no.14 in misc. ClearURLs cleans: no.2, no.6, no.7, no.8, no.11

Again, I'm not an expert on this but I'm just asking so I can get more information

edit: Just realized I referenced other issues on accident, I thought I had to select the issue when using the hashtag symbol. Sorry about that...

**Comment by atomGit** [comment]
i never actually tested CleanURLs, so i'm glad you did - given your findings, i'll have to reconsider Clean Links which is what i used before

**Comment by atomGit** [comment]
i made the mistake of writing CleanURLs instead of ClearURLs in this thread

anyway, i visited the test page you linked to and most of the samples are redirects ... ClearURLs is designed to remove tracking params, so i'm not sure if it's supposed to deal with redirects??? seems like it should be though

Skip Redirect caught all the redirect samples, but ClearURLs did not catch all of the "no redirection, only parameters" samples -- i'm not sure what to think, but maybe ClearURLs isn't the best solution - ima gonna chat with da dev n c whts up

**Comment by Thorin-Oakenpants** [comment]
https://addons.mozilla.org/firefox/addon/idn-warner/

Might be handy for those who use non-Latin, although I'm not following my logic very well here. If a russian (cryllic?) FF user **doesn't** set `network.IDN_show_punycode`=`true`, because otherwise a lot of their website addresses (using legit cryllic? characters) are confusing, then all this would do is tell them what they already know/can-see

Might be a better tool for those who do have `network.IDN_show_punycode`=`true`. IDK, i'm confusable.

**Comment by Atavic** [comment]
Repo [here](https://github.com/Mirclus/idn_warner).

**Comment by crssi** [comment]
@jingofett 

There was some problems with Clean Links way back, I do not remember anymore what and when.

ClearURL you cannot control and if something doesn't work the the only what you can do is to turn off filtering or disable extension. I do not like this extension due to lack of manual control.
The second problem is that on some pages even disabling filtering does not help and you need to disable whole extension. Unfortunately I cannot report those problem, since those pages are "private" or "secure" matter. But for example some listings does not work, that is parts of web pages on [Advanced Threat Protection](https://www.microsoft.com/en-us/WindowsForBusiness/windows-atp) configuration sites.

I use two extensions for those purposes:
1. Skip Redirect with [blacklist](https://raw.githubusercontent.com/crssi/Firefox/master/Skip_Redirect-Blacklist.txt).
2. Neat URL with [Blocked Parametes](https://raw.githubusercontent.com/crssi/Firefox/master/Neat_URL-Blocked_parameters.txt) and the following URL request types: `font, image, imageset, main_frame, media, object, object_subrequest, script, stylesheet, sub_frame, websocket, xbl, xml_dtd, xmlhttprequest, xslt, other`

Since you mentioned [https://github.com/tumpio/requestcontrol/wiki/Testing-links](https://github.com/tumpio/requestcontrol/wiki/Testing-links) test page, it passes all except 8, 13 and 14 and here is why:
8: Skipping Redirects as this sample leads to whole bunch of breakages on reddit. But you can remove `reddit.com/` from Skip Redirect blacklist (if its there). In that case tumpio example no.8 will pass.
13: Generally removing `sid` parameter from URL will also lead to breakages, but you can add `sid@mozillazine.org` to Neat URL Blocked Parameters and tumpio example no.13 will pass.
14. Allmost same as 13 (someone would need and someone not), add `dl@dropbox.com`

**Comment by atomGit** [comment]
> There was some problems with Clean Links way back, I do not remember anymore what and when.

i think it injected JS into pages - dunno if it still does

**Comment by atomGit** [comment]
[Service Worker Control by jingyu9575](https://addons.mozilla.org/en-US/firefox/addon/service-worker-control/)

> Service Workers are website scripts that run in the background, to provide features such as offline experiences, push notifications and periodic background synchronizations. By default, Firefox allows websites to register Service Workers silently, and some websites register them for unclear or undesirable purposes.
This extension detects the Service Workers registered on the pages, and allows the user to see their URL/scope or unregister them with an address bar button.
You can restrict the Service Worker feature by enabling the "Require user consent for new Service Worker registrations" option. New Service Workers will not be in effect until you click "allow" under the address bar button.

**Comment by atomGit** [comment]
suggest a small edit to wiki [4.1 Extensions](https://github.com/ghacksuserjs/ghacks-user.js/wiki/4.1-Extensions) in order to clarify things

i realize there's "We are also not saying you have to use all these extensions." but i think more clarification would be good

for example, Neat URL and ClearURLs are both listed - i might suggest listing similar extensions like these as a sub-heading under "use only one" or something

same for Skip Redirect and Redirector, and Request Control and uBO/uM

window.opener be gone - add note: not needed if "Protect window API" is enabled in CanvasBlocker

would suggest adding HTTPZ? who doesn't like kitties!

**Comment by earthlng** [comment]
Thanks @atomGit 

> Neat URL and ClearURLs

I don't use either of those so IDK if they're mutually exclusive

> Skip Redirect and Redirector

never used Skip Redirect but Redirector allows you to configure custom redirect rules whereas I believe Skip Redirect just does its thing automatically, doesn't it?

> Request Control and uBO/uM

I use all 3 of these and don't see why this would need a "use only one" header or something like that.
I guess it depends on what you want to use Request Control for.

Maybe someone who uses more of these than me can chime in here and explain the pros and cons and why one should or shouldn't use some or all of these pairs in combination

also pinging @Thorin-Oakenpants since he's the wiki master

**Comment by atomGit** [comment]
yes, Skip Redirect and Redirector should not have been lumped together - my mistake - i edited my comment to reflect that

**Comment by Thorin-Oakenpants** [comment]
I don't want to start grouping similar features, because that means maintaining it: and the overlap is not always there: depends on user's settings of said extensions, and extensions changing what they handle. The permutations are too complex

**Comment by atomGit** [comment]
all foss, all on github...

[FireMonkey](https://addons.mozilla.org/en-US/firefox/addon/firemonkey/) - handles user scripts and CSS - apparently very privacy friendly (i know there are concerns with some other script add-ons) - dev appears to be active

[Hermitation](https://addons.mozilla.org/en-US/firefox/addon/hermitation/) - toggles FPI per-domain (experimental) - 1 version released 6 months ago, so there's that

[Service Worker Detector](https://addons.mozilla.org/en-US/firefox/addon/service-worker-detector/) - interesting tool - not updated in the last year

[Enforce Browser Fonts](https://addons.mozilla.org/en-US/firefox/addon/enforce-browser-fonts/) - toggle `browser.display.use_document_fonts` - what's interesting about this to me is that the dev is planning to incorporate a list so toggling is automatic by domain

**Comment by KOLANICH** [comment]
>toggles FPI per-domain (experimental) 

As it is said in the disclaimer, FPI is global and there are no guarantees.

per-tab FPI may be in scope of https://bugzilla.mozilla.org/show_bug.cgi?id=1553791

**Comment by atomGit** [comment]
[Luminous: JavaScript events blocker](https://addons.mozilla.org/en-US/firefox/addon/luminous/)

> An extension to identify, analyze and block code execution and event collection through JavaScript in your browser.

it offers granular control over JS and can supplement other extensions, such as uBO, NS, etc.

**Comment by atomGit** [comment]
just an FYI regarding Site Bleacher (wasn't sure where best to post this)

those using ff v70 + Site Bleacher (other storage cleaners may be affected as well) may notice seemingly random crashes

the tl;dr fix is to set `dom.storage.next_gen` to false

the longer read is here...
[crashing Firefox v70 · Issue #11 · wooque/site-bleacher · GitHub](https://github.com/wooque/site-bleacher/issues/11#issuecomment-547428791)

**Comment by Thorin-Oakenpants** [comment]
The issue is https://bugzilla.mozilla.org/show_bug.cgi?id=1591374 and https://bugzilla.mozilla.org/show_bug.cgi?id=1592136

**Comment by atomGit** [comment]
dunno if this is useful (for wiki?), but i moved the uBlock O config stuff out of my Firefox config guides and consolidated it here...

[uBlock Origin Suggested Settings](https://12bytes.org/articles/tech/ublock-origin-suggested-settings)

**Comment by Solomon1732** [comment]
Looks potentially useful. Definitely not a must. Maybe a nice-to-have.
https://www.ghacks.net/2019/12/23/protect-your-tabs-in-firefox-with-dont-touch-my-tabs-relnoopener/

**Comment by Thorin-Oakenpants** [comment]
@earthlng knows more about this than me
- see https://github.com/ghacksuserjs/ghacks-user.js/issues/401#issuecomment-385262334

so we have
- a script on the wiki page (just updated)
- earthlng's `window opener be gone` extension
- don't touch my tabs rel noopener extension
- firefox's pref which is still default false - https://bugzilla.mozilla.org/show_bug.cgi?id=1522083

So working backwards...
- Firefox's pref will get flipped when they're ready - I think it has too much breakage at the moment - so forget that
- `don't touch my tabs` only modifies `<a>` tags - see link earlier - but they might not be the case now? IDK - so forget that
- earthling's extension covers things `don't touch my tabs` doesn't/didn't (and I don't mean same domain) - but it's a straight foward blocker with no whitelisting - so forget that
- The script, just rewritten by earthlng to work in greasemonkey and firemonkey (not sure if it works in violentmonkey) is the best solution: it allows you to blanket block everything but allow exceptions

tl;dr: use our script :)

**Comment by Thorin-Oakenpants** [comment]
[reddit](https://old.reddit.com/r/pihole/comments/ejvc7r/why_is_the_firefox_extension_self_destroying/fd2k9at/?context=3) .. btw, system33- works on Tor for the us navy research something something

Anyway, just thought I'd link that since I get so many messages about "cookie" addons - which 1) are not effective IMO (depends on your threat model/usage) especially when a superior solution already exists and 2) **vet** your fucking extensions people

PS: I'm not interested in cookie extesnsions, and I do not vet them ... but this one, Self-Destroying Cookies [according to the linked discussion] collects your data and is owned by Seychelle Media - which is a data collection and ad-targeting company. Not sure if anyone in the past has recommended it, but if they did, now this post is here as a record for them

**Comment by atomGit** [comment]
is font fingerprinting addressed by RFP?

[Firefox 52: Better Font Fingerprinting Protection - gHacks](https://www.ghacks.net/2016/12/28/firefox-52-better-font-fingerprinting-protection/)

[Security/Fingerprinting - MozillaWiki](https://wiki.mozilla.org/Security/Fingerprinting#Fonts)

**Comment by rusty-snake** [comment]
https://browserleaks.com/fonts seems to have no differences between RFP on and off.

**Comment by atomGit** [comment]
same here - i also tested and got the same ID - i closed the tab and disabled RFP and tested again because i thought maybe the FP would only change once you leave the domain, but alas, it didn't

**Comment by crssi** [comment]
^^ Interesting, since here I have tested right now and RFP enabled or disabled every reload same checksums when add-on is not active (disabled or not installed).
But with add-on enabled (RFP enabled or disabled) then every reload different checksum... so it seems that it does the job.

**Comment by Thorin-Oakenpants** [comment]
> is font fingerprinting addressed by RFP?

No. The only thing they did was [create a pref](https://bugzilla.mozilla.org/1121643) for whitelisting, as the first step to actually [providing](https://bugzilla.mozilla.org/1336208) a whitelist and bundling fonts. This last step will not happen if ever [I will not say why, so don't ask]

**Comment by geeknik** [comment]
Indeed. They dropped the priority of [Bundle and whitelist fonts when privacy.resistFingerprinting = true](https://bugzilla.mozilla.org/show_bug.cgi?id=1336208) from P1 to P3 a couple of years ago and nothing has happened since. And whilst [Block user-installed fonts by default](https://bugzilla.mozilla.org/show_bug.cgi?id=1582687) is new, I wouldn't expect much. Mozilla laying off 70+ employees, including some, if not all, of their QA leads sure doesn't help the mission either.

**Comment by Thorin-Oakenpants** [comment]
It's not a case of doing the work (sure, 7% of staff gone is not great) - with the removal of xul, legacy extensions etc, the ability to make rapid and lots of changes is probably unprecedented. I wish I could find it, but there was a story/quote from a Mozilla engineer about how making changes (in the past) could take 2 years to complete due to breakage: those same changes today would be 2 or 3 release cycles at most. In fact it wouldn't take much to upstream TB's patches/bundled fonts.

The reason is that the Tor Uplift is for Tor Browser's benefit (but there is another longer term goal) - and TB is already sorted out with a whitelist etc (sure, it's not perfect). So when you go to the Mozilla heads and ask for resources for this, it's just not a priority, at all really. And if you did, then you have to look at the overhead of bundling fonts (package sizes: I don't think using kinto is the answer). And you'd also have the font people asking why fonts break or look different for Japanese etc - we're probably OK with western languages using arial or times new roman: but when you look at languages with intricate script (if that's the right word), it gets messy/no-concensus.

And, it's not a perfect solution: there is still entropy in Windows in TB for font names for example, and I can guarantee the same happens in Linux. So it really all needs some data first. Even Henri's proposal is useless until they get data. And that's not to mention other font FPing: such as being able to determine the version of the font

PS: I tested that font extension quite some time ago - and I wouldn't touch it with a barge pole. There's just something about the dev and his other extensions that makes me scream inside

**Comment by crssi** [comment]
> There's just something about the dev and his other extensions that makes me scream inside

I couldn't express my feelings better.

**Comment by atomGit** [comment]
> There's just something about the dev and his other extensions that makes me scream inside

why? his other extensions are largely privacy focused

what bothers me is that it's hosted on mybrowsedaddon.com - that said, i looked at the code for Font Fingerprint Defender and there's nothing in it that sets off alarms for me, however i'm not a qualified developer

**Comment by crssi** [comment]
I don't know, but this `mybrowsedaddon.com` is the thing and not using public git makes me fell un-trusty. But is just a feeling, not that I would know some facts, might be totally unfair.

**Comment by RustyBurton** [comment]
Maybe kkapsner will be able to port the code from Font Fingerprint Defender to its extension? It would be cool.

**Comment by Thorin-Oakenpants** [comment]
No, it's not the host or the intent - it's the execution. Don't get me wrong, I'm not a dev, maybe he's learning .. IDK .. but read some reviews on all his extensions .. and in my testing the font extension failed: badly .. really badly - and no, I'm not using browserleaks. It's also that he makes multiple extensions of the same nature instead of a single one. The whole thing just smacks of something and is an instant turn-off for me. Additionally .. from a source I respect (and I won't say who) .. the code is not up to scratch

**Comment by atomGit** [comment]
thanks for expanding on your earlier comment - that's enough for me to dump it i think

**Comment by atomGit** [comment]
> Maybe kkapsner will be able to port the code ...

he decided against this a while back, but i brought it up again given that he may be able to benefit from this add-on...

[Block fonts fingerprinting · Issue #200 · kkapsner/CanvasBlocker](https://github.com/kkapsner/CanvasBlocker/issues/200)

**Comment by atomGit** [comment]
[kkapsner says...](https://github.com/kkapsner/CanvasBlocker/issues/200#issuecomment-578158039)

> You are right not to trust this extension... it's very easy to detect and does not really protect against the fingerprinting (e.g. instead of offset... the fingerprinter could just use client...).
> 
> As like other extensions regarding fingerprinting they often only protect against the proof of concept that is available in public but do not think about other attack vectors... and with fonts there are just too many. I think it's better to do this upstream.

**Comment by travankor** [comment]
**Ping Blocker**
https://github.com/dessant/ping-blocker

Ping Blocker is a browser extension that prevents sites from tracking the links you visit through hyperlink auditing.

**Comment by Thorin-Oakenpants** [comment]
looks pretty redundant and useless to me
- `browser.send_pings` already controls hyperlink auditing and is default false
- `beacon.enabled` already controls the beacon API
- uBO can already control CSP

**Comment by rusty-snake** [comment]
ClearURLs also does this since 1.11.0.

**Comment by crssi** [comment]
I believe that ghacks-user.js does this from the start, no extension is needed since then.

**Comment by atomGit** [comment]
LocalCDN - a Decentraleyes fork supporting more CDNs and it has a quick rule generation function for uBO/uM

[gitlab](https://gitlab.com/nobody42/localcdn)
[AMO](https://addons.mozilla.org/en-US/firefox/addon/localcdn-fork-of-decentraleyes/)

**Comment by travankor** [comment]
https://krebsonsecurity.com/2020/03/the-case-for-limiting-your-browser-extensions/

**Comment by crssi** [comment]
@atomGit 
LocalCDN is very promising, but due to SOP it does a lot of breakages... at lest here.
I hope he will find a solution to this problem.

**Comment by atomGit** [comment]
SOP?

**Comment by crssi** [comment]
**S**ame **O**rigin **P**olicy

See: [Broken JavaScript or CSS on some websites](https://gitlab.com/nobody42/localcdn/-/wikis/Broken-JavaScript-or-CSS-on-some-websites)

Issue: [Broken page @ codetwo.com](https://gitlab.com/nobody42/localcdn/-/issues/13#note_304369566)
[](url)

[Another example which is broken with LocalCDN](https://docs.microsoft.com/en-us/exchange/disable-or-delete-a-mailbox-exchange-2013-help)

Cheers

**Comment by atomGit** [comment]
i c - thanks

and Decentraleyes didn't have this issue?

also it seems a little disappointing maybe that he's closing those tickets - you pointed him to POOP where i'm guessing you had a similar problem and solved it, but he seems to be kicking the issue upstairs to Moz

**Comment by Thorin-Oakenpants** [comment]
FYI:
- [1551301](https://bugzilla.mozilla.org/show_bug.cgi?id=1551301)
- [1632990](https://bugzilla.mozilla.org/show_bug.cgi?id=1632990)

landed 6hrs ago (sorry for being slow on this one, I was sleeping)

**Comment by Atavic** [comment]
:+1: for [indicate tls](https://github.com/jannispinter/indicatetls)

aka SSLeuth revived.

**Comment by FintasticMan** [comment]
Would [TOS;DR](https://addons.mozilla.org/firefox/addon/terms-of-service-didnt-read/) be a good fit for this? It's an extension that grades a website's privacy policy and terms of service so that you don't have to read it yourself.

**Comment by Thorin-Oakenpants** [comment]
Not really. I'm focused on what makes the browser and web sites **actually** more private, less trackable, less fingerprintable etc with regard to web content.

How often do you even need to know any TOS - I mean, do you like sign up for new services every hour? How does it get the TOS? Does it connect to a third party? Does it do this automatically, or do you click a button when you want the info? IDK - not even going to check it out, sorry.

Just harden your browser, learn to compartmentalize, and treat **every service as fucking evil by default** when you weigh up whether or not you want to use them, because they are all fucking evil, or liars, or have shit security, use double-speak or newspeak, change terms on a whim, etc when it comes to your data.

**Comment by atomGit** [comment]
[Service Worker Detector by Thomas Steiner](https://addons.mozilla.org/en-US/firefox/addon/service-worker-detector/) - [github](https://github.com/google/service-worker-detector)

**Comment by gwarser** [comment]
@atomGit https://github.com/google/service-worker-detector/issues/13

**Comment by atomGit** [comment]
CSS Exfil Protection bug

test: https://coreboot.org/status/board-status.html

CSS Exfil Protection causes sustained high CPU for the 'Web' process and very slow page loading (may not complete at all)

tested with ff 77.0.1 on linux, clean (default) profile

[issue link](https://github.com/mlgualtieri/CSS-Exfil-Protection/issues/26)

**Comment by travankor** [comment]
https://github.com/mindedsecurity/behave

**Behave!** monitors and warn if a web page performs any of following actions:

- Browser based Port Scan
- Access to Private IPs
- DNS Rebinding attacks to Private IPs

**Comment by atomGit** [comment]
[http tracker by Venu Madhav](https://addons.mozilla.org/en-US/firefox/addon/http-tracker/) - potentially interesting for you developer types i suppose - i don't know that it offers anything more than the developer tools, but the layout might be nicer

**Comment by atomGit** [comment]
another possibly useful one for testing perhaps...

[Moesif Origin & CORS Changer](https://addons.mozilla.org/en-US/firefox/addon/moesif-origin-cors-changer1/)

**Comment by EchoDev** [comment]
[Cookie AutoDelete](https://addons.mozilla.org/firefox/addon/cookie-autodelete/) [(Git)](https://github.com/Cookie-AutoDelete/Cookie-AutoDelete) has been updated to support removing cache, indexDB, local storage, plugin data, service workers and cookies per host. 

Is Cookie Autodelete missing important data that has to be cleaned? Afaik this cleans all possible data set by the website in Firefox. If the cleaning is complete this should be recommended.

**Comment by atomGit** [comment]
> Is Cookie Autodelete missing important data that has to be cleaned?

it cleans iDB, but not on tab close (host leave) - i thought Moz updated the API to allow cleaning per-host on host leave, but apparently not ???

eTag gets dumped with cache - i think with FPI enabled and some cache dumping, eTags are not really an issue

**Comment by atomGit** [comment]
ps. i use [Site Bleacher](https://addons.mozilla.org/en-US/firefox/addon/site-bleacher/) - this is the only one i'm aware of that auto-dumps iDB while surfing, though it dumps it globally, not per-domain

**Comment by EchoDev** [comment]
> it cleans iDB, but not on tab close (host leave) - i thought Moz updated the API to allow cleaning per-host on host leave, but apparently not ???

Are you sure about this? I can configure cleaning IndexedDB per host and I can't find anything in the release notes that says otherwise? From my own test it seems to clean iDB from hosts that I told CAD to clean just fine.

**Comment by atomGit** [comment]
CAD cleans iDB on browser exit (or start) i believe and it may clean per-domain (i didn't test) but according to my tests it doesn't auto-clean iDB per-domain after you x the tab - i confirmed this by looking in about:preferences#privacy at the stored data and there was always 64KB of storage that never got dumped (after tab close)

funny thing is, i could not see evidence of this data in the dev tools -> storage tab

are you seeing something different?

[What is the state of the WebExtension API to clean X data?](https://github.com/Cookie-AutoDelete/Cookie-AutoDelete/wiki/FAQ:-Common-Questions-and-Issues#what-is-the-state-of-the-webextension-api-to-clean-x-data)

**Comment by EchoDev** [comment]
> CAD cleans iDB on browser exit (or start) i believe and it may clean per-domain (i didn't test) but according to my tests it doesn't auto-clean iDB per-domain after you x the tab - i confirmed this by looking in about:preferences#privacy at the stored data and there was always 64KB of storage that never got dumped (after tab close)

This seems to be an empty database created by Firefox. It sometimes leaves a caches.sqlite file in %firefox profile%\storage\default\%site-name%\cache folder.

I checked the sqlite file and it is db with identical tables but no data (I can't find why Firefox creates this?)

If you check the iDB data that actually gets created, CAD does delete it. The actual data gets created in %firefox profile%\storage\default\%site-name%\idb folder

 [What is the state of the WebExtension API to clean X data?](https://github.com/Cookie-AutoDelete/Cookie-AutoDelete/wiki/FAQ:-Common-Questions-and-Issues#what-is-the-state-of-the-webextension-api-to-clean-x-data)

The CAD wiki is outdated since service workers etc. are supported.

**Comment by atomGit** [comment]
thank you for digging deeper than i did - i kind of thought it odd when i saw the screen-cap where every storage item was exactly 64KB which i thought might be, like you said, an empty db - this explains why i wasn't seeing any storage data in dev tools

**Comment by kennethtran93** [comment]
Thanks for the further research.  I'll update our docs with your findings :D

**Comment by Thorin-Oakenpants** [comment]
> i confirmed this by looking in about:preferences#privacy at the stored data and there was always 64KB of storage that never got dumped (after tab close)
> ...
> funny thing is, i could not see evidence of this data in the dev tools -> storage tab

inspector and the info in about:preferences#privacy Manage Data (quota manager?) are not always in sync - there are lots of open bugs about this sort of thing - including a never removed 2kb. When doing testing, I have always opened and looked in the sqlite files. IDB is a bit harder - this is why I created https://ghacksuserjs.github.io/TorZillaPrint/sanitizing.html to help me, over a year ago - sorry, but I never got round to adding service worker cache

---

It helps (overall internal FF sanitizing and perf and other things - maybe it also helps extensions get the correct info in time, IDK) if you have Next Gen Local Storage enabled (I have) - it's more robust and doesn't rely (as much?) on arbitrary/wonky asynchronous timings. Even little things such as shared workers across tabs (same domain) reporting the same info

`dom.storage.next_gen` = true
- this was going to be enabled for ESR68, but got pulled/reversed at the last minute. It was then targeted as ESR78 but missed the boat. AFAIK the reason has to do with current quota data, storage versions, old data corruption or whatever. So if you clear all your shit, then turn it on you guys :) I've had it on for about the last 8 releases.

**Comment by atomGit** [comment]
i think [Extension source viewer](https://addons.mozilla.org/en-US/firefox/addon/crxviewer/) would be worthwhile adding to the [Extensions \ Tools](https://github.com/arkenfox/user.js/wiki/4.1-Extensions#small_orange_diamond-extensions-tools) part of the wiki

**Comment by EchoDev** [comment]
@atomGit it is already in the wiki.
CRX Viewer = Extension source viewer

**Comment by atomGit** [comment]
i... um... i knew that - just wanted to see if i could catch you off guard :)

**Comment by syncng** [comment]
Just letting y'all know
If there so happens to be a comment on [this page](https://12bytes.org/articles/tech/firefox/firefox-extensions-my-picks) in particular referring to this event with my name on it, **it's totally not me**

**Comment by atomGit** [comment]
> I can't make a comment there ...

i saw your 3 comments - first-timers have to be manually approved :)

**Comment by atomGit** [comment]
[nuTensor](https://github.com/geekprojects/nuTensor) - appears to be an active uMatrix fork - not on AMO yet, but apparently will be

**Comment by Thorin-Oakenpants** [comment]
I don't think uMatrix is needed anymore TBH.. I get that the matrix (per type per scope) and interface for that is awesome, but uBO can do almost anything uM can - I plan to properly explore this in #1044 . As long as uM isn't broken, I think it's ok to use it, as long as users are aware of any pitfalls/alternatives

At this stage, I wouldn't touch imitators/forks until it has a proven record and superior really needed features

**Comment by atomGit** [comment]
> I don't think uMatrix is needed anymore

i think you might be right, and obviously Gorhill thinks the same, but i'm having a lot of trouble wrapping my head around that (course it is 2020 where nothing makes sense to me) - i've tried disabling uM and using uBO in advanced mode **many** times and i always end up re-enabling uM ...... guess i'll give it another try because i would like to dump uM

BTW, uBO can do per-scope dynamic filtering too, it's just more of a pain in the sub-abdominal sphincter region, not that i would bother

**Comment by atomGit** [comment]
appears nearly identical to uMatrix, but a new 'fetch' filter has been added to the pop-up UI and in the 'more' tab...

```
assetFetchBypassBrowserCache false
assetFetchTimeout 30
autoUpdateAssetFetchPeriod 120
```

[LibreMatrix on github](https://github.com/LibreMatrix/LibreMatrix)
[LibreMatrix on AMO](https://addons.mozilla.org/en-US/firefox/addon/librematrix/)

**Comment by geeknik** [comment]
I thought `fetch` was added to the latest unreleased uMatrix code.  Doesn't feel new. 

> appears nearly identical to uMatrix, but a new 'fetch' filter has been added to the pop-up UI and in the 'more' tab...

**Comment by atomGit** [comment]
> I thought fetch was added to the latest unreleased uMatrix ...

that may be so - i was comparing to the AMO release

he has made several commits though

**Comment by heretoenhance** [comment]
May I propose to add config suggestions for the listed extensions? For example CanvasBlocker has a ton of settings and I am sure I'm not the only one that feels a little scared to "break" the setup by changing some, even though the default ones could not be "enough" (e.g. protect window API is off by default), another example is ClearURLs that has host blocking on by default but, given that uBlock Origin is suggested, they would interfere, same thing with again ClearURLs (and its Filter ETAGs headers from requests) with Header editor or ETag Stoppa.

As it emerged also from what I wrote above, another good addition to the extensions page on the wiki would be to group extensions by their functions in order to avoid conflicts (ClearURLs/ NeatURL) sort of like what is already done with Header Editor and ETag Stoppa.

As last thing, what do you guys think of NoScript with all restrictions off by default so that it only works to "Sanitize cross-site suspicious requests" setiing in the Advanced Setting panel? I am not using uMatrix and I don't know if this would be useless with it.

**Comment by crssi** [comment]
Where would ClearURLs (CUs) interfere with uBlock Origin (uBO)?
About ETags... CUs, HE and ETag Stoppa do not interfere but just share/duplicate some of functionality.
If you use ETag removal in CUs, then you dont need ETag Stoppa or HE to remove ETags, but nothing will break if you do... it just only the first in a row will remove it and others will have nothing to remove anymore.
ATM I am using CUs without ETag removal and use ETag Stoppa for handling ETags, just because CUs adds to counter also all ETags removal actions, which is almost every page, so I can+t see exactly when CUs did some link cleaning beside ETags removal action... so I have separated those.

CUs and NeatURL are similar but also very different.
CUs is a set and forget, which is cool, BUT without any option to control and if there is a breakage you need to report it and pray to god it will be resolved soon, and AFAICT sometimes its done within a few weeks and sometimes within a few months... see here https://gitlab.com/KevinRoebert/ClearUrls/-/issues... for example this one https://gitlab.com/anti-tracking/ClearURLs/rules/-/issues/115 was opened 3 weeks ago and nothing was done about yet.
On the other hand NeatURL gives you better control, but it also gives you more work to fine tune.
I don have other opinions about those... ATM I am using CUs and ETags Stoppa.

NoScript is at "No bother" section here https://github.com/arkenfox/user.js/wiki/4.1-Extensions#small_orange_diamond-dont-bother ATM. I do not use it.

Cheers

**Comment by heretoenhance** [comment]
With interfere I meant duplicate funcionality which I guessed is not something one would want, otherwise for example why not using multiple Adblockers? CUs has domain blocking which is redundant if using uBlock Origin, please correct me if I'm wrong.

Again I didn't mean to say that using CUs default with HE and/or ETag Stoppa would break something, just that is pointless to duplicate funcionality.

I know the wiki suggests those extensions but it doesn't say to install all of them, that is obvious; what I was saying is that grouping them and suggesting configs would make things more clear for everyone, especially newbies.

Finally I know NoScript is on the "No bother" section but I was asking about a particular use of it and not its main, so my question was if that function was considered at all.

**Comment by EchoDev** [comment]
You dont want multiple adblockers because for example some adblockers inject their own neutered script to make the site think Google Analytics is working. There is a chance they mess up stuff for each other.

When removing ETags you dont input anything new. So once the ETag has been removed the other extension will just do nothing. That is why you can chain then without issues.

**Comment by rusty-snake** [comment]
> CUs has domain blocking which is redundant if using uBlock Origin, please correct me if I'm wrong.

1. UBO has much more complete filterlists, CUs has only some common domains.
2. CUs can only block full domains IIRC, while UBO can block trackers and ads from a domain and allow articles and images.

>  NoScript is on the "No bother" section …

… but w/o arguments why it is there (since the end of the CSP game). Newbies (and I) may ask what are the reason.

**Comment by heretoenhance** [comment]
I was obviously not arguing against uBlock Origin which IMO is one of the best privacy tools out there. I was just making a point that the wiki would benefit from suggesting configs, like disabling Domain blocking on CUs since it is redundant and other things

**Comment by rusty-snake** [comment]
Newbies would strongly welcome config suggestions, that's true. FYI: [my config suggestions](https://gitlab.com/rusty-snake/firefox-config/-/blob/master/LOG.md). targeting advanced users; w/o explanation.

**Comment by atomGit** [comment]
@heretoenhance - i keep [2 Firefox config guides](https://12bytes.org/articles/tech/firefox/) if you want to have a look, as well as a [uBO guide](https://12bytes.org/articles/tech/firefox/ublock-origin-suggested-settings/) which covers different scenarios

just know that i do not claim to be more knowledgeable than others here and i tend to do things a little different sometimes in the interest of less problematic web browsing

**Comment by theltalpha** [comment]
I want to second @atomGit's advise: His guides _Firefox Configuration Guide for Privacy Freaks and Performance Buffs_ and _uBlock Origin Suggested Settings_ are most useful. Until recently (commit 9f99885) @atomGit's a.k.a 12bytes user-overrides.js and Firefox guides were also recommended by arkenfox. It will take some hours of very concentrated work to work through and comprehend the guides and some hours to carefully examine the user-overrides.js which might need some adjustment to fit your needs, but everything is written very clear and comprehensible, and once this is done, you just have to look out for modifications and comments in the guides, which happen every few weeks and are not too massive, and commits to the user-overrides.js, which also do not happen too often.

By the way, why was the reference to 12bytes and his guides removed in the README.md?

**Comment by handshake900** [comment]
Happy holidays! I hope…

I’ve been off the map for a while and haven’t used my machine since FFv83. Just wanted to see if you guys are still using uM and/or if it plays well or is necessary with FF84. Same with UBO. I checked out 12org and there was a small change in regard to hyperlink auditing.

Thanks.

**Comment by atomGit** [comment]
some are, some aren't - personally i still use it and yes, it's plays fine with v84 and probably will continue to do so for some time

i'd like to dump uM and stick with just uBO, but that doesn't quite work for me - uBO can do most of what uM does, but then you're in a text editor writing rules all the time

**Comment by seonwoolee** [comment]
Same here - I know I can manipulate uBO to do what UM does, but the interface of uM is just so much faster and easier to use

**Comment by Duckbilled** [comment]
If this is the wrong place for my question please let me know. 

I went through the extensions and found Smart Referer. I was checking a bit what it was and I do not 100% understand it yet. But, checking [wikipedia](https://en.wikipedia.org/wiki/HTTP_referer#Referer_hiding), I found the following line: 
> "If a website is accessed from a HTTP Secure (HTTPS) connection and a link points to anywhere except another secure location, then the referrer field is not sent." 

Does this mean that if you have HTTPS-only, the extension does not really matter?

**Comment by EchoDev** [comment]
@Duckbilled it does matter. The referrer is still sent if you go from a https to https site. The bit you quoted only matters if you go from https to http

Tbh if you follow the settings from arkenfox user.js you referrer options are pretty strict. Smart Referer is more usefull if you didn't set your referrer options through the user.js

**Comment by B00ze64** [comment]
FYI "LibreMatrix" has disappeared. Hopefully someone else forks uM,,,

**Comment by eric15342335** [comment]
should i use cookie-autodelete?

**Comment by Thorin-Oakenpants** [comment]
> should i use cookie-autodelete?

Are you using first party isolation? If you are, then what is your threat model? Clearing cookies (and **hopefully** other related data like service worker caches, IDB, localStorage etc) only makes sense if you're trying to stop tracking repeat visits to the same site **_within_** a session (you are sanitizing on close, right?) .. and even then, your visits are still linkable via your IP (so I would assume you have a VPN and it changes from time to time, and of course that temp IP you're using is shared)

tl;dr: what is your threat model or end-game?

**Comment by rawlife56** [comment]
@Thorin-Oakenpants I have a question, does using extensions like Dark reader or Dark background and light text increase entropy ? Can they be detected by the website as we are making changes to the design locally ?

I'm specifically using Dark background and light text with a slightly tweaked grey color as the website background. Will this make me stand out in any obvious way ?

**Comment by rusty-snake** [comment]
If javascript is enabled, yes.

### PoC

Open a page (I used wikipedia), open the console and execute
```js
window.getComputedStyle(document.getElementsByTagName("body")[0]).backgroundColor
```

With Dark Reader: "rgb(29, 32, 33)"
Without Dark Reader: "rgb(246, 246, 246)"

**Comment by Thorin-Oakenpants** [comment]
If you modify the dom, then it is detectable: checking a single attribute like body would be simple and fast for FP script

**Comment by atomGit** [comment]
i would guess such scripts are not widespread and that those being used are not parsing through all the attribs.

that said, i wonder if the server could hash the document when it spits it out, then compare the hash (assuming JS) with a client-side hash?

**Comment by rawlife56** [comment]
> If javascript is enabled, yes.

Strange, mine shows the exact background color value that I set even with the Java script disabled using Ublock origin. I tried changing the values and executing this resulted in whatever values I set every time. 
I guess using default values which came with the extension makes more sense so that i will at least fall in a bucket of users using ublock and dark mode extension?

**Comment by rusty-snake** [comment]
> even with the Java script disabled using Ublock origin

Disabling JS in uB does not affect JS entered by the user in the console.

**Comment by rawlife56** [comment]
> > even with the Java script disabled using Ublock origin
> 
> Disabling JS in uB does not affect JS entered by the user in the console.

Oh! my bad. Missed that logic.

**Comment by jsalomons** [comment]
Really grateful to have found your wiki and user.js - thanks for all your good work!

I'd really appreciate your opinion on LocalCDN. It's a fork of Decentraleyes. Is it worth using?

**Comment by DanKGooGLy** [comment]
Why don't you guys refer https://github.com/Sainan/Universal-Bypass instead of https://github.com/sblask/webextension-skip-redirect?

**Comment by atomGit** [comment]
@DanKGooGLy - correct me if wrong, but Universal Bypass seems to be a very different animal than Skip Redirect in that UB doesn't skip redirects

**Comment by tirphana** [comment]
Could anyone provide an opinion/recommendation concerning https://addons.mozilla.org/en-US/firefox/addon/trackmenot/, considering it’s no longer being maintainted?

**Comment by g-2-s** [comment]
For uMatrix the wiki says "Use it as long as it works for you... except that's risky, because how do you know it's working properly?". As far as I can see (which is not much considering I'm no expert), it seems to work quite efficiently still, but am I missing some crucial detail here? I'd be glad to ditch it for uB0 only but uM is simpler to use in my case.

**Comment by atomGit** [comment]
[Port Authority by ACK-J](https://addons.mozilla.org/en-US/firefox/addon/port-authority/) 

github: https://github.com/ACK-J/Port_Authority

> Blocks websites from using javascript to port scan your computer/network and dynamically blocks all LexisNexis endpoints from running their invasive data collection scripts.

not sure this is something worth using - feedback appreciated

**Comment by curiosityseeker** [comment]
> [Port Authority by ACK-J](https://addons.mozilla.org/en-US/firefox/addon/port-authority/)
> 
> github: https://github.com/ACK-J/Port_Authority
> 
> > Blocks websites from using javascript to port scan your computer/network and dynamically blocks all LexisNexis endpoints from running their invasive data collection scripts.
> 
> not sure this is something worth using - feedback appreciated

Doesn't this add-on offer what [gwarser's lan-block.txt list](https://github.com/gwarser/filter-lists/blob/master/lan-block.txt) already provides?

@gwarser : What do you think?

**Comment by potassiumchloride** [comment]
> For uMatrix the wiki says "Use it as long as it works for you... except that's risky, because how do you know it's working properly?". As far as I can see (which is not much considering I'm no expert), it seems to work quite efficiently still, but am I missing some crucial detail here? I'd be glad to ditch it for uB0 only but uM is simpler to use in my case.

I'm still using uMatrix (uM) in hardcore mode, too, in combination with uBlockOrigin (uBO), and just like you I prefer uM to handle javascript per site basis. I find its user interface much more convenient, and it allows better control than uBO. As far as I can say in my daily usage, it still works perfectly (just to give you some feedback from another user, @g-2-s)! Anyway, uM+uBO is still the perfect combination/solution for me.

However, I don't really acknowledge the very biased description of @Thorin-Oakenpants about uM in the Wiki. It's nothing but pure FUD and should be replaced with a neutral, objective comment (i.e. just saying that it's currently unmaintained and nothing else!). I disliked this change of subjective personal wording from the very beginning, when uM was transferred into the _Extensions-maybe_ section, but I'm a bit late with my complaint now. :-)

**Comment by kah0922** [comment]
> 
> 
> > [Port Authority by ACK-J](https://addons.mozilla.org/en-US/firefox/addon/port-authority/)
> > github: https://github.com/ACK-J/Port_Authority
> > > Blocks websites from using javascript to port scan your computer/network and dynamically blocks all LexisNexis endpoints from running their invasive data collection scripts.
> > 
> > 
> > not sure this is something worth using - feedback appreciated
> 
> Doesn't this add-on offer what [gwarser's lan-block.txt list](https://github.com/gwarser/filter-lists/blob/master/lan-block.txt) already provides? It blocks the scans on https://defuse.ca/in-browser-port-scanning.htm
> 
> @gwarser : What do you think?

uBlock Origin's CNAME blocking also takes care of the LexisNexis endpoint blocking as well.

On another note, has anyone checked out https://github.com/garywill/autoreferer?

Also with AdGuard URL Tracking filter being added to uBlock Origin, Neat URL is redundant.

**Comment by Gitoffthelawn** [comment]
> On another note, has anyone checked out https://github.com/garywill/autoreferer?

It looks good, but I prefer tools that allow you to specify the referer depending on the source URL and/or target URL, not the tab/window.

**Comment by Thorin-Oakenpants** [comment]
I'm going to quote potassiumchloride's from minimized comment three posts up

> However, I don't really acknowledge the very biased description of @Thorin-Oakenpants about uM in the Wiki. It's nothing but pure FUD and should be replaced with a neutral, objective comment (i.e. just saying that it's currently unmaintained and nothing else!). I disliked this change of subjective personal wording from the very beginning, when uM was transferred into the Extensions-maybe section, but I'm a bit late with my complaint now. :-)

... and then I'm going to FUCKING RUB IT IN HIS FACE

Not cheering the fact this happened to uM, just pointing out that my apparent "very biased" "subjective" "personal" "FUD" was anything but

![uM](https://user-images.githubusercontent.com/16656956/125796848-ca546fb9-fb97-4477-ac75-68c70a34dde4.png)

**Comment by rusty-snake** [comment]
FYI: https://github.com/gorhill/uMatrix/releases/tag/1.4.2

**Comment by B00ze64** [comment]
hpHosts has not disappeared from my uMatrix, even though I updated to 1.4.2 just now, unchecked the list, and restarted the browser. No harm done, this was just a quick edit to the default config I see on the commit...

**Comment by g-2-s** [comment]
I updated to 1.4.4 but the "Reveal canonical names" option is now gone and adding the rule manually does nothing, can anyone confirm if they have the same issue?

**Comment by EchoDev** [comment]
> 
> 
> I updated to 1.4.4 but the "Reveal canonical names" option is now gone and adding the rule manually does nothing, can anyone confirm if they have the same issue?

Get this build if you want cname uncloaking https://github.com/gorhill/uMatrix/releases/tag/1.4.3b0
cname uncloaking never made it to stable (it also has a DNS leak issue when using SOCKS proxy)

**Comment by Solomon1732** [comment]
uBO-Scope is abandonware. Latest commit was in July 2018
https://github.com/gorhill/uBO-Scope/commits/master

**Comment by Thorin-Oakenpants** [comment]
> uBO-Scope is abandonware. Latest commit was in July 2018. Might want to at least take note of it in the Extensions page.
> https://github.com/gorhill/uBO-Scope/commits/master

IDK if it's abandoned. Maybe it doesn't need anything done to it - @gorhill - then again, it could probably do with cname detection to properly ascertain partyness? the rest seems perfectly fine. Hopefully gorhill will reply

**Comment by practik** [comment]
I guess this is still the right place for extension talk, even though it's closed?

[A while back](https://github.com/arkenfox/user.js/issues/294#issuecomment-398273478) there was some discussion of [Universal Bypass](https://addons.mozilla.org/en-US/firefox/addon/universal-bypass/), an extension for demystifying shortened links. claustromaniac checked it out and found it was no help in the privacy dept.

I generally avoid those links but every so often I encounter one (e.g. in an e-mail from the water company) that I have to click. So I looked to see what else is available on AMO these days and found four: [Fast Forward](https://addons.mozilla.org/en-US/firefox/addon/fastforwardteam/) (a fork of Universal Bypass), [Link Unshorten](https://addons.mozilla.org/en-US/firefox/addon/link-unshorten/), [Quo Vadis?](https://addons.mozilla.org/en-US/firefox/addon/quo-vadis/), and [Unshort.link](https://addons.mozilla.org/en-US/firefox/addon/unshort-link/).

I am no 🐈 but I did my best to test them out, and it looks to me like all but Quo Vadis? work by sending your URL to the developer's server or a third-party service for analysis, which I don't love.

Quo Vadis? stumped me: It sends requests to the shortened URL and the final URL and any URLs in between, but none of those requests show up in the browser console; I could only see them by monitoring my computer's network traffic. So I wrote to the developer (https://basa.nl/quovadis/), who answered:

> Quo Vadis? works by using the browser's built-in ability to perform HTTP requests. It runs entirely within the browser. When it is triggered by the user, it launches its own private (sandboxed) HTTP-request for the initial URL. This does not involve any user oriented "browsing", all data is absorbed internally, nothing is rendered. It then asks the browser to inform the extension of any redirections that occur for that request. This information includes the next URL, and the next, etc. That's it!

"Sandboxed" definitely sounds good, but I don't have the skills to evaluate what that really means, so I'm posting here in case others are interested in looking into it further.

**Comment by lhindir** [comment]
I’d appreciate more justification for Skip Redirect, since it’s the only extension to share the Recommended tier with uBlock Origin. Most of the Optional tier have caveats, and I think Skip Redirect does as well. While it works well when it works, it applies to a small enough class of links that I don’t think it’s worth the recommendation.

Inherently, it can only detect trivial redirect links directly including the target URL. base64-encoding the target is enough to get past it. Many sites work this way; email newsletter service tracking links come to mind. Similarly, any custom server-side mapping of ID’s to target links, e.g. any URL shortener service, goes undetected. This can give users a false sense of security since Skip Redirect only skips the most benign redirect links.

Additionally, it has a number of false positives (e.g. archive.org) that require users to attempt to exhaustively enumerate exceptions. This is futile, especially when services change their URL schemes.

Given the prevalence of both false negatives and false positives, I’m not a big fan and don’t use it myself. Obviously, the wiki is just recommendations, so it’s up to the author and doesn’t force anyone to follow it, but since I think most of the rest of the repo and wiki are very well-thought-out, I’m genuinely curious about the rationale here.

**Comment by xe-3** [comment]
> I’d appreciate more justification for Skip Redirect, since it’s the only extension to share the Recommended tier with uBlock Origin. Most of the Optional tier have caveats, and I think Skip Redirect does as well. While it works well when it works, it applies to a small enough class of links that I don’t think it’s worth the recommendation.

I also think providing further explanation/justification for Skip Redirect would provide value.

**Comment by ntnguyen1234** [comment]
> base64-encoding the target is enough to get past it.

I think Skip Redirect can go to base64-encoded link. I just tested with this link `http://example.com/?url=aHR0cHM6Ly93aWtpcGVkaWEub3JnLw==` and it redirects successfully for me (no `example.com` connection made in network console).

**Comment by lhindir** [comment]
Yeah, I was wrong about that. The rest of my comment still applies.

**Comment by stephenhawk8054** [comment]
Btw, uBO has added a new network filter [`urlskip`](https://github.com/gorhill/uBlock/commit/266ec4894b) to skip click trackers. It does not skip automatically like "Skip Redirect" though, but based on the filters in the list:

https://github.com/uBlockOrigin/uAssets/blob/ea7bea1f5a86cbf32291c4b166b8c0d198b92273/filters/privacy.txt#L579-L773

**Comment by lhindir** [comment]
Yeah, urlskip works great. Confirmed in the browser console that it skips the redirect URLs YouTube puts in video descriptions now.

@Thorin-Oakenpants thoughts on dropping Skip Redirect given that there’s actively maintained uBO filters accomplishing the same goal? It’s not dynamic, but that’s also a nice reduction in extension surface for the average user (particularly since it’s another manifest v2 extension)
